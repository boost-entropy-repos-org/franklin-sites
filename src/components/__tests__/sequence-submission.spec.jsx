import React from 'react';
import { render, fireEvent, cleanup, waitForElement } from '@testing-library/react';
import SequenceSubmission from '../sequence-submission';

afterEach(cleanup);

describe('SequenceSubmission', () => {
  test('should render', () => {
    const { asFragment } = render(<SequenceSubmission />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render with sequence is too short error', () => {
    const { asFragment } = render(<SequenceSubmission value="ACTG" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render with sequence is invalid error', () => {
    const { asFragment } = render(<SequenceSubmission
      value="ACTGUACTGUACTGU+"
    />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render with sequence is missing error', () => {
    const { asFragment } = render(<SequenceSubmission value="  " />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should validate after onChange event', async () => {
    const { queryByTestId } = render(<SequenceSubmission />);
    const textarea = queryByTestId('sequence-submission-input');
    fireEvent.change(textarea, { target: { value: 'ACTG' } });
    const error = await waitForElement(() =>
      queryByTestId('sequence-submission-error')
    );
    expect(error).toBeTruthy();
  });

  test('should call custom onChange callback once', async () => {
    const onChange = jest.fn();
    const value = 'A';
    const { queryByTestId } = render(<SequenceSubmission
      onChange={e => onChange(e)}
    />);
    const textarea = queryByTestId('sequence-submission-input');
    fireEvent.change(textarea, { target: { value } });
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(value);
  });
});
