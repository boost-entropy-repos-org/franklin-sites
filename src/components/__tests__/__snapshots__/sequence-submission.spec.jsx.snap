// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SequenceSubmission should render 1`] = `
<DocumentFragment>
  <textarea
    class="sequence-submission-input"
    data-testid="sequence-submission-input"
  />
</DocumentFragment>
`;

exports[`SequenceSubmission should render with sequence is invalid error 1`] = `
<DocumentFragment>
  <textarea
    class="sequence-submission-input"
    data-testid="sequence-submission-input"
  >
    ACTGUACTGUACTGU+
  </textarea>
  <div
    class="message message--failure"
    data-testid="sequence-submission-error"
    role="status"
  >
    <div
      class="message__side-border"
    />
    <test-file-stub
      height="1.125em"
      width="1.125em"
    />
    <section
      class="message__content"
    >
      <small>
        The sequence is invalid
      </small>
    </section>
  </div>
</DocumentFragment>
`;

exports[`SequenceSubmission should render with sequence is missing error 1`] = `
<DocumentFragment>
  <textarea
    class="sequence-submission-input"
    data-testid="sequence-submission-input"
  >
                
  </textarea>
  <div
    class="message message--failure"
    data-testid="sequence-submission-error"
    role="status"
  >
    <div
      class="message__side-border"
    />
    <test-file-stub
      height="1.125em"
      width="1.125em"
    />
    <section
      class="message__content"
    >
      <small>
        The sequence is missing
      </small>
    </section>
  </div>
</DocumentFragment>
`;
